### PR TITLE
refactor: [M3-7585] - Upgrade to React 18

### PR DESCRIPTION
**Pull request by bnussman-akamai**
## Description 📝

Updates to React 18 📦

This PR does everything needed to update to React 18 and the latest React testing library. 🎉

I really wanted to break up this work into smaller PR, but the lack of backwards compatibility with React Testing made this approach more realistic.  😖

## Main Changes 🔁
- Updates many unit tests to `await` React Testing Library's `userEvent` calls ⏱️
- Rewrites a few unit tests that involve Action Menus to be more verbose and straight forward - this relates to the `userEvent` change 🖊️
- Removes `jest-axe` because we aren't really using it 🗑️ (I'm not opposed to using it, I just think we should revisit this and add it if we're gonna actually commit to using it. It was only used on one test)
- Refactored unit tests that test React hooks because `@testing-library/react-hooks` no longer exists. You now have to use `renderHook` from `@testing-library/react` 🪝
- Uses a yarn resolution to pin `jackspeak` version to fix cypress / yarn / storybook issue 📌 (see https://github.com/cypress-io/cypress/issues/27370#issuecomment-1652953961) 
- The native `<select>` no longer allows `placeholder` as a prop, so you will see changes to account for this ⌨️

## How to test 🧪

- Verify all automated testing passes ✅
- Verify the app works as expected when running locally and in the preview link
  - We obviously can't all test every little feature, but spend 30 min or so clicking around Cloud Manager and look for bugs 🔎
  
## As an Author I have considered 🤔

- [ ] 👀 Doing a self review
- [ ] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [ ] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a changeset
- [x] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [ ] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

File: packages/manager/src/components/Uploaders/ImageUploader/ImageUploader.tsx [hunk 1]
```diff
@@ -216,7 +217,11 @@ export const ImageUploader = React.memo((props: ImageUploaderProps) => {
           variant: 'success',
         });
 
-        dispatchAction(setPendingUpload(false));
+        // React force a render so that `pendingUpload` is false when navigating away
+        // from the upload page.
+        flushSync(() => {
+          dispatchAction(setPendingUpload(false));
+        });
```
Comment (abailly-akamai) [inline packages/manager/src/components/Uploaders/ImageUploader/ImageUploader.tsx:224]: This is new to me. Could we use `const { rerender } = renderWithTheme` instead? Not too important, this just looks odd

**Review by abailly-akamai — APPROVED** [review]
Thanks for this major update & cleanup (and all the work preceding it)!

I spent quite a bit of time navigating and all over the APP and so far sop good (I will spend more time and we'll have plenty time to test this in develop).

The only thing I noticed is that cold loading behaves a bit differently, seems to take a bit more time to resolve API requests maybe, or for the update to be reflected in the UI?). It often results in visible layout shifts. I can't quite put my finger on it yet. (will spend more time on it and try while serving a production build). It may also just be my own distorted impression.

Meanwhile the build job seems much faster 🎉

**Review by mjac0bs — APPROVED** [review]
Thanks for completing this effort! 🎉 

The amount of times you had to type/find-and-replace `await` in this PR is... so many. 😅 🥇 

I spent a while going through the app, checking functionality of several product create flows (linodes, kube, firewall, volume, obj) and details pages, plus the accounts pages, checking a variety of different components and comparing with prod. Not seeing any regressions and I couldn't seem to replicate what Alban was seeing with longer API req resolution times, though we should keep an eye on that in develop as a team.

If we plan on merging this right after the next release branch is cut, we'll have a good amount of time for this to sit in develop.

**Review by cpathipa — APPROVED** [review]
Nice work. Thank you @bnussman! Over all LGTM! No regression found in core features 
(Linode / Volume / OBJ / Kubernetes / VPC and firewalls).

**Pull request merged**